File: frontend/src/app/features/dashboard/application/application-dto.ts
import {ApplicationState} from '../../../shared/application-state';

export interface ApplicationDto {
  id: number;
  email: string;
  fullName: string;
  phone: string;
  companyName: string,
  taxNumber: string,
  source: string,
  comment?: string,
  state: ApplicationState,
  salesId?: number,
  salesName?: string,
  referralCode?: string,
  createdAt: string,
  modifiedAt: string
}

File: backend/src/main/java/hu/ibc/ibcpartners/security/dto/ApplicationDto.java
package hu.ibc.ibcpartners.security.dto;

import hu.ibc.ibcpartners.security.entity.ApplicationState;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

import java.time.Instant;

public record ApplicationDto(
        Long id,
        String email,
        String fullName,
        String phone,
        String companyName,
        String taxNumber,
        String source,
        String comment,
        ApplicationState state,
        Long salesId,
        String salesName,
        String referralCode,
        Instant createdAt,
        Instant modifiedAt
) {}
